ttnn/cpp/ttnn/operations/eltwise/binary/device/broadcast_height_multi_core_sharded_optimized_program_factory.cpp

```diff
@@ -91,7 +91,7 @@ BinaryDeviceOperation::BroadcastHeightMultiCoreShardedOptimized::create(
     TT_FATAL(input_tile_size == output_tile_size, "Input and output tile size should be same");
     uint32_t shard_size_in_bytes = shard_spec.numel() * a.element_size();
```

Comment:
@eyonland not directly related to this issue but this calculation for shard size doesn't seem correct for data types like bfp8. I saw we assert that a is tile layout so bfp8 is a valid dtype, and element_size() returns 1 for bfp8. Ex if the shard was 1 tile this would calculate 1024B, but bfp8 tile size is 1088B.